akka-projection-core/src/main/scala/akka/projection/StatusObserver.scala

```diff
@@ -31,10 +31,10 @@ abstract class StatusObserver[-Envelope] {
   def stopped(projectionId: ProjectionId): Unit
 
   /**
-   * Called when the envelope has been processed and corresponding offset stored.
+   * Called when the corresponding offset has been stored (even if the envelope has not been processed yet).
    * It might not be called for each envelope.
    */
-  def progress(projectionId: ProjectionId, env: Envelope): Unit
+  def offsetProgress(projectionId: ProjectionId, env: Envelope): Unit
 
```

Comment:
Change in semantics: this method will no longer wait for the event to be processed too. This reports progress of offsets.